File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    length = len(prices)
    
    for i in range(0, length, 1):
        flag = False
        for j in range(i, length, 1):
            if(prices[i]>prices[j] or i == length-1):
                answer.append(j-i)
                flag = True
                break
        if(flag == False):
            answer.append(length-i-1)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    var answer = [...new Array(prices.length)].map(()=>0);
    const stack = [];
    let time = -1;
    
    prices.forEach((price,i)=>{
        time ++;
        while(stack[stack.length-1] && price<stack[stack.length-1][0]){
            let idx = stack.pop()[1];
            answer[idx] = time-idx;
        }
        stack.push([price,i]);
    })
    
    stack.forEach(value=>answer[value[1]]=time-value[1]);

    return answer;
}